Products/ZenModel/BasicDataSource.py

```diff
@@ -38,6 +38,39 @@ def checkOid(oid):
     return oid
 
 
+class SnmpCommand(object):
+    '''
+    Builds the command for SNMP.i  v3 has additional arguments
+    while v1/v2 just use the string template.
+    '''
+    def __init__(self, snmpinfo):
+        self.snmpinfo = snmpinfo
+
+    def getCommand(self):
+        command = snmptemplate % self.snmpinfo
+
+        if self.snmpinfo['zSnmpVer'] != 'v3':
+            return (command, command)
+
+        # v3 always requires the username
+        command += (" -u%(zSnmpSecurityName)s" % self.snmpinfo)
+        display = command
+
+        if self.snmpinfo['zSnmpPrivType'] and self.snmpinfo['zSnmpAuthType']:
+            display += (" -l authPriv -a %(zSnmpAuthType)s " % self.snmpinfo) + "-A ${zSnmpAuthPassword} " + \
+                ("-x %(zSnmpPrivType)s " % self.snmpinfo) + "-X ${zSnmpPrivPassword}"
+            command += (" -l authPriv -a %(zSnmpAuthType)s -A %(zSnmpAuthPassword)s "
+                "-x %(zSnmpPrivType)s -X %(zSnmpPrivPassword)s" % self.snmpinfo)
+        elif self.snmpinfo['zSnmpAuthType']:
+            display += (" -l authNoPriv -a %(zSnmpAuthType)s " % self.snmpinfo) + "-A ${zSnmpAuthPassword}"
```

Comment:
I was replicating the output of the Command->smptwalk output (which is also
broken for v3), but can change it to ***** if that's better?  I'm
indifferent on which to use

On Mon, Nov 30, 2015 at 10:14 AM, jplouis <notifications@github.com> wrote:

> In Products/ZenModel/BasicDataSource.py
> <https://github.com/zenoss/zenoss-prodbin/pull/1268#discussion_r46163579>:
>
> > +        command = snmptemplate % self.snmpinfo
> > +
> > +        if self.snmpinfo['zSnmpVer'] != 'v3':
> > +            return (command, command)
> > +
> > +        # v3 always requires the username
> > +        command += (" -u%(zSnmpSecurityName)s" % self.snmpinfo)
> > +        display = command
> > +
> > +        if self.snmpinfo['zSnmpPrivType'] and self.snmpinfo['zSnmpAuthType']:
> > +            display += (" -l authPriv -a %(zSnmpAuthType)s " % self.snmpinfo) + "-A ${zSnmpAuthPassword} " + \
> > +                ("-x %(zSnmpPrivType)s " % self.snmpinfo) + "-X ${zSnmpPrivPassword}"
> > +            command += (" -l authPriv -a %(zSnmpAuthType)s -A %(zSnmpAuthPassword)s "
> > +                "-x %(zSnmpPrivType)s -X %(zSnmpPrivPassword)s" % self.snmpinfo)
> > +        elif self.snmpinfo['zSnmpAuthType']:
> > +            display += (" -l authNoPriv -a %(zSnmpAuthType)s " % self.snmpinfo) + "-A ${zSnmpAuthPassword}"
>
> Should the passwords be show as "*****" instead of "${zSnmpAuthPassword}"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zenoss/zenoss-prodbin/pull/1268/files#r46163579>.
>